### PR TITLE
Simplify Glyph

### PR DESCRIPTION
**Pull request by madig**
Closes https://github.com/linebender/norad/issues/92.

I'm keeping `Outline` around for only-once checking. Maybe... I can cut down the builders again and get the only-once logic back into the glif parser...

### CODE REVIEW

**Comment by madig** [comment]
I am now looking at de-Option-ing the libs of Anchor, Guideline, Contour, ContourPoint and Component. There, we have the added headache of identifiers. The `replace_lib` method takes care to assign an identifier.. but what if we don't have an Option to guide us? Auto-assign in `lib_mut` even when nothing is added?

**Comment by cmyr** [comment]
Is the option in these types a big problem, ergonomics-wise? I think that creating an identifier on access is reasonable, it's sort of like 'copy on write' semantics.

**Comment by madig** [comment]
Since there doesn't seem to be much code accessing those libs, I can't say, it _is_ very slightly annoying when you have to, though. Maybe I'll leave it for now...

**Comment by madig** [comment]
I wonder why a release build is bigger now :thinking:

File: src/glyph/builder.rs [hunk 1]
```diff
@@ -271,6 +282,12 @@ pub struct OutlineBuilder {
     scratch_state: OutlineBuilderState,
 }
 
+#[derive(Debug, Clone, Default, PartialEq)]
+pub struct Outline {
```
Comment (cmyr) [inline src/glyph/builder.rs:286]: If this is now purely an internal type, we can remove the `pub`.

**Review by cmyr — APPROVED** [review]
One little note, otherwise this looks good to me!

File: src/glyph/builder.rs (hunk 1 above)
Comment (madig) [reply src/glyph/builder.rs:286]: Hm, then I can't use it in parse.rs :thinking:

Comment (madig) [reply src/glyph/builder.rs]: Removed it from the public API.

**Pull request merged**